spec/controllers/application_controller_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+RSpec.describe ApplicationController do
+  controller do
+    def set_seller_test
+      set_seller
+    end
+
+    def require_seller_login_test
+      require_seller_login
+    end
+  end
+
+  describe "require_seller_login" do
+    before :each do
+      @seller1 = Seller.create(username: "user1", email: "email1@email.com", password: "password1", password_confirmation: "password1")
+      routes.draw { get "require_seller_login_test" => "anonymous#require_seller_login_test" }
+    end
+
+    context "seller not logged in" do
+      before :each do
+        get :require_seller_login_test, session: { seller_id: nil }
+      end
+
+      it "sets flash[:errors] to include :not_logged_in" do
+        expect(flash[:errors]).to include { :not_logged_in }
+      end
+
+      it "redirects to login_path" do
+        expect(response).to redirect_to login_path
+      end
+    end
+
+    context "seller logged in" do
+      before :each do
+        get :require_seller_login_test, session: { seller_id: @seller1.id }
+      end
+
+      it "does not set flash[:errors] to include :not_logged_in" do
+        expect(flash[:errors]).to_not include { :not_logged_in }
+      end
+    end
+  end
+
+  describe "set_seller" do
+    before :each do
+      @seller1 = Seller.create(username: "user1", email: "email1@email.com", password: "password1", password_confirmation: "password1")
+      @seller2 = Seller.create(username: "user2", email: "email2@email.com", password: "password2", password_confirmation: "password2")
```

Comment:
I don't think the `@seller2` is needed in before each since it appears as though it is only used in one of the tests below